### ISSUE TITLE
"geckodriver unexpectedly exited" error with 0.10

### ISSUE THREAD

**Issue by MarkUK** [mozilla/geckodriver#184]
I get the following trace whenever I try to use geckodriver 0.10 via a minimal Python script (attached):

```
======================================================================
ERROR: setUpClass (__main__.SampleManagerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "minimal.py", line 15, in setUpClass
    cls.driver = webdriver.Firefox(capabilities=caps, executable_path='geckodriver-0.10')
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/firefox/webdriver.py", line 65, in __init__
    self.service.start()
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/common/service.py", line 86, in start
    self.assert_process_still_running()
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/common/service.py", line 99, in assert_process_still_running
    % (self.path, return_code)
WebDriverException: Message: Service geckodriver-0.10 unexpectedly exited. Status code was: 1
```

This does not occur with geckodriver 0.90. I have checked the following:

* geckodriver-0.10 is in a directory that's in the PATH, and can be executed directly.
* The download appears to be sound. The "file" command reports that geckodriver-0.10 is an ELF binary, as I would expect.
* "pip" reports that the latest version of the Selenium libraries is installed.

Further information:

* I'm running Firefox Developer Edition (currently v50).
* I'm using Python 2.7
* The OS is Ubuntu Mate 16.04
* Modifying the test file to point to geckodriver-0.9 works fine.
* I'm running the test from a command line using "python minimal.py" with no other parameters.

Any help or suggestions would be greatly appreciated.

[minimal.py.zip](https://github.com/mozilla/geckodriver/files/422153/minimal.py.zip)

**Comment by andreastt** [comment]
What is the stdout/stderr from the geckodriver binary? Are you able to execute the file fine?

**Comment by MarkUK** [comment]
I don't see anything on the terminal other than the message above, when running geckodriver via Python.

I've tried running it directly using:

    geckodriver-0.10 -b '/path/to/firefox'

but I don't see any output (or any Firefox) at all. The program appears to be running in the process list (and has gained a --webdriver-port argument), which I guess is the expected behaviour. The same occurs when I run 0.9 in this way.

If there's another way to run the executable, or some other arguments I should be passing to it, please let me know. Thanks.

**Comment by andreastt** [comment]
It would be useful if you ran the binary with the `-vv` flag to get more verbose output. I suspect your test runner above is swallowing the stdout/stderr output, however.

**Comment by gentoo90** [comment]
I have the same trouble when running
```python
from selenium import webdriver
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
caps = DesiredCapabilities.FIREFOX.copy()
caps['marionette'] = True
br = webdriver.Firefox(capabilities=caps, executable_path='./target/debug/geckodriver')
```
After debugging python I found out that command line for `geckodriver` is
```bash
./target/debug/geckodriver -b /usr/bin/firefox --webdriver-port 45753
```
which gives
```
error: Found argument '--webdriver-port' which wasn't expected, or isn't valid in this context

USAGE:
    geckodriver --binary <BINARY>

For more information try --help
```

**Comment by gentoo90** [comment]
Looks like already fixed in SeleniumHQ/selenium#2544

**Comment by andreastt** [comment]
Ah, that makes sense. This reminds me that it would be nice if Selenium could propagate the error message when it exits with a non-zero exit code.
